src/main/java/gg/projecteden/nexus/models/vaults/VaultUser.java

```diff
@@ -0,0 +1,69 @@
+package gg.projecteden.nexus.models.vaults;
+
+import dev.morphia.annotations.Converters;
+import dev.morphia.annotations.Entity;
+import dev.morphia.annotations.Id;
+import gg.projecteden.mongodb.serializers.UUIDConverter;
+import gg.projecteden.nexus.framework.exceptions.postconfigured.InvalidInputException;
+import gg.projecteden.nexus.framework.interfaces.PlayerOwnedObject;
+import gg.projecteden.nexus.framework.persistence.serializer.mongodb.ItemStackConverter;
+import gg.projecteden.nexus.utils.PlayerUtils;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static gg.projecteden.utils.StringUtils.plural;
+
+@Data
+@Entity(value = "vault_user", noClassnameStored = true)
+@NoArgsConstructor
+@AllArgsConstructor
+@RequiredArgsConstructor
+@Converters({UUIDConverter.class, ItemStackConverter.class})
+public class VaultUser implements PlayerOwnedObject {
+	@Id
+	@NonNull
+	private UUID uuid;
+	private Map<Integer, List<ItemStack>> vaults = new HashMap<>();
+	private int limit;
+
+	public static final int ROWS = 4;
```

Comment:
aren't player vaults supposed to have 6 rows?

Reply:
Currently unused (well used but has no affect), new vaults will have a close button and pagination, meaning 4 rows.

Reply:
not seeing any code that will handle the reduction from 6 rows per vault to 4

Reply:
I haven't written it. This is a direct copy of the implementation from the plugin, future me will handle the reduction (like moving overflow)